File: scripts/update.lua
local cjson = require("cjson")

local function read_file(path)
    local file = io.open(path, "r")
    if not file then
        return nil
    end
    local content = file:read("*a")
    file:close()
    return content
end

local function convert(t)
    local formatted = {}
    -- Iterate over gitmojis
    for _, v in pairs(t.gitmojis) do
        table.insert(formatted, {
            value = v.emoji,
            text = v.code,
            description = v.description,
        })
    end
    return formatted
end

local function dump(obj)
    if type(obj) == "table" then
        local s = "{ "
        for k, v in pairs(obj) do
            if type(v) == "string" then
                v = '"' .. v .. '"'
            end
            -- print(type(k), k, v)
            s = s .. k .. " = " .. dump(v) .. ","
        end
        return s .. "}"
    else
        return tostring(obj)
    end
end

local contents = read_file("/tmp/gitmojis.json")
local t = convert(cjson.decode(contents))

local function todo(x)
    local output = "local emojis = {"
    for k, v in pairs(x) do
        output = output .. dump(v) .. ","
    end
    output = output .. [[
}

return emojis
    ]]
    return output
end

print(todo(t))

File: scripts/update.py
import sys
import json
import pathlib
from typing import Dict, List


def read_file(file: pathlib.Path):
    with file.open() as fd:
        content = json.load(fd)
    return content.get("gitmojis", [])

def write_file(file: pathlib.Path, content: str):
    with file.open("w") as fd:
        fd.write(content)


def to_lua(emojis: List[Dict[str, str]]):
    s = "local emojis = {\n"
    for item in emojis:
        value = item.get("emoji")
        text = item.get("code")
        description = item.get("description")
        s += f'{{ value = "{value}", text = "{text}", description = "{description}"}},'
    s += "}\n\nreturn emojis"
    return s


def main(infile: str, outfile: str):
    emojis = read_file(pathlib.Path(infile))
    write_file(pathlib.Path(outfile), to_lua(emojis))


if __name__ == "__main__":
    main(sys.argv[1], sys.argv[2])
